pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -542,9 +543,9 @@ func routeSemanticEquals(desired, existing *v1alpha3.VirtualService) bool {
 func getHostPrefix(isvc *v1beta1.InferenceService, disableIstioVirtualHost bool) string {
 	if disableIstioVirtualHost == true {
 		if isvc.Spec.Transformer != nil {
-			return constants.DefaultTransformerServiceName(isvc.Name)
+			return constants.TransformerServiceName(isvc.Name)
```

Comment:
We need to add a check if the transformer or predictor exists with the `default` suffix.

Reply:
we need an e2e test for disabling the top level istio virtual host

Reply:
I am confused if the original bug is for serverless mode or raw deployment mode

Reply:
This bug is only valid for serverless deployment with network layer other than istio

Reply:
But I think this problem is only for the master branch not in the release right ?

Reply:
yes, It is introduced when we removed default suffix

Reply:
@yuzisun Tested it with istio and kourier. It is working fine :)